### PR TITLE
chore: addon generator to async/await

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
refactor

**Did you add tests for your changes?**
yes

**If relevant, did you update the documentation?**
NA

**Summary**
* Move async gen to async await.

**Does this PR introduce a breaking change?**

No

**Other information**
Not yet

### CODE REVIEW

File: packages/generators/addon-generator.ts [hunk 1]
```diff
@@ -39,12 +39,9 @@ export default function addonGenerator(
 		public copy: (value: string, index: number, array: string[]) => void;
 		public copyTpl: (value: string, index: number, array: string[]) => void;
 
-		public prompting(): Promise<{}> {
-			return this.prompt(prompts).then(
-				(props: InquirerScaffoldObject): void => {
-					this.props = props;
-				}
-			);
+		public async prompting(): void {
```
Comment (ematipico) [inline packages/generators/addon-generator.ts]: As the test said, it should return `Promise<void>`. Strange, the IDE the have told you

Comment (anshumanv) [reply packages/generators/addon-generator.ts]: Strange too. 🤔 

Resolving.

File: packages/generators/addon-generator.ts [hunk 2]
```diff
@@ -39,12 +39,9 @@ export default function addonGenerator(
 		public copy: (value: string, index: number, array: string[]) => void;
 		public copyTpl: (value: string, index: number, array: string[]) => void;
 
-		public prompting(): Promise<{}> {
-			return this.prompt(prompts).then(
-				(props: InquirerScaffoldObject): void => {
-					this.props = props;
-				}
-			);
+		public async prompting() {
```
Comment (ematipico) [inline packages/generators/addon-generator.ts]: I don't know if you have rebased from master, but now with eslint we force defining a return type. So please use `Promise<void>`

Comment (anshumanv) [reply packages/generators/addon-generator.ts]: 😮 something definitely is wrong with my editor.

Fixed this, thanks for your patience on this @ematipico ✨

**Pull request merged**